packages/block-library/src/social-link/edit.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { URLPopover } from '@wordpress/block-editor';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	IconButton,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getIconBySite } from './social-list';
+
+const SocialLinkEdit = ( { attributes, setAttributes, isSelected } ) => {
+	const { url, site } = attributes;
+	const [ showURLPopover, setPopover ] = useState( false );
+	const classes = classNames(
+		'wp-social-link',
+		'wp-social-link-' + site,
+		{ 'wp-social-link__is-incomplete': ! url },
+	);
+
+	// Import icon.
+	const IconComponent = getIconBySite( site );
+
+	return (
+		<Button
+			className={ classes }
+			onClick={ () => setPopover( true ) }
+		>
+			<IconComponent />
+			{ isSelected && showURLPopover && (
+				<URLPopover
+					onClose={ () => setPopover( false ) }
+				>
+					<form
+						className="block-editor-url-popover__link-editor"
+						onSubmit={ ( event ) => {
+							event.preventDefault();
+							setPopover( false );
+						} } >
+						<div className="editor-url-input block-editor-url-input">
```

Comment:
Class names are not meant to be reused outside the component which defines them. In this case, we should go through the abstraction of the `URLInput` component, not use its classes directly. Or, alternatively, copy all of its styles to a class specific to the Social Link. These guidelines are meant to ensure that styles only apply within the context of a component. Otherwise, we risk future breakage on changes expected only to affect `URLInput`. 

https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#naming

>A component's class name should **never** be used outside its own folder (with rare exceptions such as `_z-index.scss`). If you need to inherit styles of another component in your own components, you should render an instance of that other component. At worst, you should duplicate the styles within your own component's stylesheet. This is intended to improve maintainability by treating individual components as the isolated abstract interface.